catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/metacard/associations/Associated.java

```diff
@@ -160,6 +162,11 @@ private void addEdge(Edge edge, Map<String, Metacard> metacards,
         String id = edge.parent.get(Metacard.ID)
                 .toString();
         Metacard target = changedMetacards.getOrDefault(id, metacards.get(id));
+
+        if (target == null) {
+            return;
+        }
```

Comment:
Why this check only on  `addEdge`? should we add it to `removeEdge` too?  
Also if that target metacard is not there that means we can't complete part of the requested update. Like before not sure if we want to fail or allow the partial update.